### DISCUSSION TITLE
go-swagger not generating yaml that matches example

### DISCUSSION THREAD

**Discussion by csphilli** [go-swagger/go-swagger#2826] [Q&A]
I'm following the model example on [goswagger.io](https://goswagger.io/use/spec/model.html).

Here is my User model including comments. It is in `package routes` 

```
// User is the model for Me
//
// swagger:model User
type User struct {
	// The name of the User
	//
	// required: true
	// min length: 3
	Name string `json:"name"`

	// The email of the user
	// required: true
	// example: john@doe.com
	Email string `json:"email"`
}
```

Here is the file structure. `main.go` is in `package main` with a reference to routes as an import statement.

```
├── src
│   ├── cmd
│   │   └── random_name
│   │           └──main.go
│   └── routes
│           └── user.go
```

To generate the spec I use `swagger generate spec -m -o ./swagger.yaml` while being in the directory of main.go.

The expected outcome is this format (from the goswagger.io example - my output context is different though). Please note the extent of content for the definitions.

```
definitions:
  User:
    type: object
    title: User represents the user for this application
    description: "A user is the security principal for this application.\nIt's also used as one of the main axes for reporting.\n\nA user can have friends with whom they can share what they like."
    required:
      - id
      - name
      - login
    properties:
      id:
        description: the id for this user
        type: integer
        format: int64
        minimum: 1
      name:
        description: the name for this user
        type: string
        minLength: 3
      login:
        description: the email address for this user
        type: string
        format: email
        x-go-name: Email
        example: user@provider.net
      friends:
        description: the friends for this user
        type: array
        items:
          $ref: "#/definitions/User"
        x-property-value: value
        x-property-array:
          - value1
          - value2
        x-property-array-obj:
          - name: obj
            value: field
```
            
And my output is:
```
definitions:
    User:
        description: User is the model for Me
        x-go-package: <import path referenced from main.go - omitted for privacy>
```
Why is go swagger outputting `x-go-package: <path>` instead of all the expected properties as in the example? Is it a problem with my generate spec command? If so, what is the actual command? If that's not it, what other areas might I check?

I am ultimately trying to use `swagger-codegen` to generate files based on the `-l typescript-axios` language. And when I pass it my swagger.yaml file, the `user.ts` file looks like this:
```
export interface User{
}
```

**Comment by casualjim** [comment]
What is the command that you run?  

Unless the model is reachable from the main function, it might not get included until you add the flag `--scan-models`

**Reply by csphilli** [reply]
I've been running `swagger generate spec -m -o ./swagger.yaml`. I've also run `--scan-models` as well and still getting the `x-go-package` information instead of actual values.

To see if there was something else wrong with my annotations, I cloned the entire go-swagger repo, and ran the `swagger spec generate --scan-models -o ./swagger.yaml` against the [petstore](https://github.com/go-swagger/go-swagger/tree/master/fixtures/goparsing/petstore) example. Even that is resulting in definitions with `x-go-name` and `x-go-package`.

Then I run `swagger-codegen generate -i swagger.yaml -l typescript-axios -o ./openAPI/`. My goal is to have `openAPI/models/user.ts` actually have data there but I only get
```
export interface User{}
```
